apm-agent-plugins/apm-opentelemetry/apm-opentelemetry-metricsdk-plugin/src/main/java/co/elastic/apm/agent/otelmetricsdk/SdkMeterProviderBuilderInstrumentation.java

```diff
@@ -0,0 +1,96 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.otelmetricsdk;
+
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.GlobalTracer;
+import co.elastic.apm.agent.otelmetricexport.ElasticOtelMetricsExporter;
+import co.elastic.apm.agent.sdk.ElasticApmInstrumentation;
+import co.elastic.apm.agent.sdk.logging.Logger;
+import co.elastic.apm.agent.sdk.logging.LoggerFactory;
+import co.elastic.apm.agent.sdk.weakconcurrent.WeakConcurrent;
+import co.elastic.apm.agent.sdk.weakconcurrent.WeakSet;
+import io.opentelemetry.sdk.metrics.SdkMeterProviderBuilder;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+public class SdkMeterProviderBuilderInstrumentation extends ElasticApmInstrumentation {
+    @Override
+    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return named("io.opentelemetry.sdk.metrics.SdkMeterProviderBuilder");
+    }
+
+    @Override
+    public ElementMatcher<? super MethodDescription> getMethodMatcher() {
+        return named("build");
+    }
+
+    @Override
+    public Collection<String> getInstrumentationGroupNames() {
+        return Arrays.asList("opentelemetry", "opentelemetry-metrics", "experimental");
+    }
+
+    @Override
+    public String getAdviceClassName() {
+        return getClass().getName() + "$SdkMeterProviderBuilderAdvice";
+    }
+
+    /**
+     * Instruments {@link SdkMeterProviderBuilder#build()}.
+     */
+    public static class SdkMeterProviderBuilderAdvice {
+
+        private static final Logger logger = LoggerFactory.getLogger(SdkMeterProviderBuilderInstrumentation.SdkMeterProviderBuilderAdvice.class);
+
+        private static final WeakSet<SdkMeterProviderBuilder> ALREADY_REGISTERED_BUILDERS = WeakConcurrent.buildSet();
+        private static final ElasticApmTracer tracer = GlobalTracer.requireTracerImpl();
+
+        @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
+        public static void onEnter(@Advice.This SdkMeterProviderBuilder thiz) {
+            if (checkMetricsSdkVersionSupported()) {
+                if (ALREADY_REGISTERED_BUILDERS.add(thiz)) {
+                    ElasticOtelMetricsExporter.createAndRegisterOn(thiz, tracer);
+                }
+            }
+        }
+
+        private static boolean checkMetricsSdkVersionSupported() {
```

Comment:
This will very likely spam the logs with warnings, thus ensuring it is only triggered once would be better.